department-of-veterans-affairs/vets-api
# Update SignUpServiceUpdaterJob to process using latest agreement

## Summary

 - Update `SignInServiceUpdaterJob` to always process the latest agreement of a given version. This is because Sign up service does not keep a record of agrees and declines so there is a possibility things could get out of order. Keep this job unique.
- Cleanup and add more error handling

## Related issue(s)
- department-of-veterans-affairs/va.gov-team#70063

## Testing
- Sign in 
- navigate to `localhost:3001/terms-of-use`
- decline terms
- Lookup user in rails console and check that they have declined `terms_of_use_agreement`
  - `UserAccount.find_by(icn: <icn>)`
- navigate to `localhost:3000/sidekiq/queues/default` and you should see your `TermsOfUse::SignUpServiceUpdaterJob` enqueued
- Repeat for accepted

## What areas of the site does it impact?
Terms of use

## Acceptance criteria
